risc0/bigint2/src/ec/mod.rs

```diff
@@ -160,31 +177,55 @@ impl<const WIDTH: usize, C: Curve<WIDTH>> AffinePoint<WIDTH, C> {
         }
 
         // Return the result, based on which buffer was written to last.
-        let result_point = if result_flip { result2 } else { result1 };
-        result.buffer = result_point;
+        *result = if result_flip { result2 } else { result1 };
     }
 
     #[stability::unstable]
     pub fn double(&self, result: &mut Self) {
         let curve = C::CURVE;
-        double_raw(self.as_u32s(), curve.as_u32s(), &mut result.buffer);
+        if self.is_zero {
+            *result = *self;
+        } else {
+            unsafe {
+                double_raw(self.as_u32s(), curve.as_u32s(), &mut result.buffer);
+            }
+            result.is_zero = false;
```

Comment:
This will work for the current curves, but if we ever do anything on a curve with points of order 2, we'll need to check if the y coordinate is zero. Perhaps something like the following:
```suggestion
            if self.buffer[1] != [0u32; WIDTH] {
                unsafe {
                    double_raw(self.as_u32s(), curve.as_u32s(), &mut result.buffer);
                }
                result.is_zero = false;
            } else {
                result.is_zero = true;
            }
```

Reply:
Sorry I read this after pushing a commit, I think https://github.com/risc0/risc0/pull/2572/commits/8b31b229f8300ccd38e2af48ff2a8617a7bde230 is that diff (maybe double check I didn't apply in the wrong place)

Should this also be done in the add fn? (it calls raw method since the is_zero already checked)

Reply:
LGTM

The add fn is good as-is. This is the analog of the `self.buffer[0] == rhs.buffer[0]` and `self.buffer[1] != rhs.buffer[1]` case over in `add` -- that case checks for `A + -A = 𝒪` (where `𝒪` is the zero point), and this case checks for when `2A = 𝒪` because that happens exactly when the y coordinate is zero (because `-(x, y)` is `(x, -y)` on elliptic curves, so `A = -A` when `y = 0`)